src/swar_synthesizer.py

```diff
@@ -4,82 +4,98 @@
 import struct
 import pyaudio
 
-def swar2freq(swar,basefreq):
+SWAR_FREQUENCY_MAPPER = {
+    's':{
+        'shuddh':1,
+        'vakra':1, # Error handling mechanism.
+    },
+    'r':{
+        'shuddh': 1.125,
+        'vakra': 1.058
+    },
+    'g':{
+        'shuddh': 1.266,
+        'vakra': 1.188
+    },
+    'm':{
+        'shuddh': 1.333,
+        'vakra': 1.412
+    },
+    'p':{
+        'shuddh': 1.5,
+        'vakra': 1.5 # Error handling mechanism
+    },
+    'd':{
+        'shuddh': 1.688,
+        'vakra': 1.586
+    },
+    'n':{
+        'shuddh': 1.898,
+        'vakra': 1.78
+    },
+    '0': 0,
+}
+VALID_SWAR_INPUTS_SUBSET_1 = set(['<', '>'])
+VALID_SWAR_INPUTS_SUBSET_2 = set(['s', 'r', 'g', 'm', 'p', 'd', 'n'])
+VALID_SWAR_INPUTS = set(
+    list(VALID_SWAR_INPUTS_SUBSET_1) + list(VALID_SWAR_INPUTS_SUBSET_2)
+)
+
+
+def swar_to_frequency(swar, base_frequency):
     """
-    Contains a dictionary (mapper) that stores relates the swar to 
+    Uses a dictionary (mapper) that relates the swar to
     its defined frequency. I have used the Pythagorean ratios
-    to define the relative frequencies based a fixed basefreq
+    to define the relative frequencies based a fixed base_frequency
     variable that is set to 440 Hz by default, but can be speci-
     fied by the User.
-    This function interprets the octave specifier which halves 
+    This function interprets the octave specifier which halves
     or doubles the value of the defined note to shift down or up
     on the scale.
-    """
-    basefreq_float=float(basefreq) # Perform type conversion once
-
-    mapper={
-        's':{
-            'shuddh':1,
-            'vakra':1, # Error handling mechanism.
-            },
-        'r':{
-            'shuddh': 1.125,
-            'vakra': 1.058
-            },
-        'g':{
-            'shuddh': 1.266,
-            'vakra': 1.188
-            },
-        'm':{
-            'shuddh': 1.333,
-            'vakra': 1.412
-            },
-        'p':{
-            'shuddh': 1.5,
-            'vakra': 1.5 # Error handling mechanism
-            },
-        'd':{
-            'shuddh': 1.688,
-            'vakra': 1.586
-            },
-        'n':{
-            'shuddh': 1.898,
-            'vakra': 1.78
-            },
-        '0':0,
-        
-    }
-
-    """
     For any other input, including captialization, this function
```

Comment:
I was hoping to use capitalization for some thing else. Not sure right now, but I might consider using capital letters for shuddh swars and small letters for komal/teevr swars.